taker-frontend/src/components/CloseButton.tsx

```diff
@@ -0,0 +1,129 @@
+export const CloseButton = ({ cfd, connectedToMaker }: CloseButtonProps) => {
+    let [settle, isSettling] = usePostRequest(`/api/cfd/${cfd.order_id}/settle`);
+    let [commit, isCommiting] = usePostRequest(`/api/cfd/${cfd.order_id}/commit`);
+
+    const disableCloseButton = cfd.state.getGroup() === StateGroupKey.CLOSED
+        || !(cfd.state.key === StateKey.OPEN);
+
+    let closeButton;
+
+    if (connectedToMaker) {
+        closeButton = <Box w={"45%"}>
+            <Popover
+                placement="bottom"
+                closeOnBlur={true}
+            >
+                {({ onClose }) => (<>
+                    <PopoverTrigger>
+                        <Button colorScheme={"blue"} disabled={disableCloseButton}>Close</Button>
+                    </PopoverTrigger>
+                    <PopoverContent color="white" bg="blue.800" borderColor="blue.800">
+                        <PopoverHeader pt={4} fontWeight="bold" border="0">
+                            Collaboratively Close your position
+                        </PopoverHeader>
+                        <PopoverArrow />
+                        <PopoverCloseButton />
+                        <PopoverBody>
+                            <Text>
+                                This will collaboritively close your position with the counterparty. The exchange rate
+                                at {cfd.expiry_timestamp}
+                                will determine your profit/losses. It is likely that the rate will change until then.
+                            </Text>
+                        </PopoverBody>
+                        <PopoverFooter
+                            border="0"
+                            d="flex"
+                            alignItems="center"
+                            justifyContent="space-between"
+                            pb={4}
+                        >
+                            <Button
+                                size="sm"
+                                colorScheme="red"
+                                onClick={() => {
+                                    console.log(`Settling CFD ${cfd.order_id}`);
+                                    settle({});
+                                    onClose();
+                                }}
+                                isLoading={isSettling}
+                            >
+                                Close
+                            </Button>
+                        </PopoverFooter>
+                    </PopoverContent>
+                </>)}
+            </Popover>
+        </Box>;
+    } else {
+        closeButton = <Box w={"45%"}>
+            <Popover
+                placement="bottom"
+                closeOnBlur={true}
+            >
+                {({ onClose }) => (<>
+                    <PopoverTrigger>
+                        <Button colorScheme={"blue"} disabled={disableCloseButton}>Commit</Button>
```

Comment:
I'd call it `Close` as I don't think users will understand the difference here. 
```suggestion
                        <Button colorScheme={"blue"} disabled={disableCloseButton}>Close</Button>
```

Reply:
won't they be puzzled if they don't get they money until the next day? IMHO having the same button with the same name will eventually lead to users being unable to accurately create a mental map of software's functionality. Perhaps at least we could color-code it differently?

Reply:
i went with close and force-close